### DISCUSSION TITLE
Proper workflow for Rails?

### DISCUSSION THREAD

**Discussion by Overload119** [planetscale/discussion#109] [FAQs] [answered]
These are currently in my notes on how to make changes to the DB. 

```
1. Create a new branch.
  1. `pscale branch create <db> <branch>`
  2. `pscale deploy-request create <db> my-branch`
2. Add a migration file. 
  1. `spring rails generate migration <name> --database <db>`
  2. Edit the migration file.
3. Execute Rails changes.
```

It seems like after a fork of a new branch, the rows in the previous branch don't get deployed.
Therefore I can't execute against them because I get the following error message:

```
AlreadyExists desc = Table '<tablename>' already exists (errno 1050) (sqlstate 42S01)
```

My master branch has the rows that should not execute the old tables that exist.

![CleanShot 2021-09-23 at 20 07 31](https://user-images.githubusercontent.com/1012072/134612089-fe02e2b8-8a34-4b30-8935-fe31be692eb2.png)

What am I doing wrong?

(I can unblock myself by inserting the schema migration manually but this is unproductive).

**Comment by mscoutermarsh** [comment, marked as answer]
Hi @Overload119,

You can enable having `schema_migrations` auto copied over for you. Go to your database settings, and select `Rails`.
<img width="675" alt="CleanShot 2021-09-27 at 12 57 01@2x" src="https://user-images.githubusercontent.com/155044/134953014-93c2b875-0bfd-450c-983f-295b4ab026b3.png">

Now each time you do a deploy request or branch, any changes to `schema_migrations` will get persisted across branches.

**Comment by Overload119** [comment]
@mscoutermarsh I actually don't get how this works - 

I already have this enabled on my `leads/main` branch. 
When I create a new branch via `pscale branch create <db> <branch>` it will not migrate over schema_migrations. 

As a result, I have to manually add them.

**Reply by mscoutermarsh** [reply]
@Overload119 Did you just recently enable it? After being enabled, it takes effect after the next deploy request (that's when the records get copied over + start getting tracked).

If you're still having trouble, certainly feel free to email support and we can check on specifics for your DB.
